### PR TITLE
feat(editor): add Kate

### PR DESCRIPTION
**Pull request by CoelacanthusHex**
- [x] Introduction
- [x] Install
- [x] Basic editor features
- [x] file encode and end of line
- [x] Search and replace (with Regex)
- [x] LSP
- [x] Terminal
- [x] External tools (e.g. 
     - [x] format with clang, rustfmt, gofmt or XML formatter 
     - [x] git blame 
     - [x] compile and run single cpp file)

### CODE REVIEW

File: docs/tools/editor/kate.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+author: CoelacanthusHex
+
+## 软件简介
+
+Kate 是一个具有众多功能的跨平台文本编辑器。Kate 还附带了多种插件，包括一个嵌入式终端，可以让你直接从 Kate 中启动控制台命令，强大的搜索和替换插件，以及一个预览插件，可以让你看到你的 MD、HTML 甚至 SVG 的样子。支持通过交换文件在系统崩溃时恢复数据，带参数提示的自动补全，同时支持 LSP 以获得更为强大的补全。
```
Comment (ksyx) [inline docs/tools/editor/kate.md:5]:
```suggestion
Kate 是一个具有众多功能的跨平台文本编辑器。Kate 还附带了多种插件，包括一个嵌入式终端，可以让你直接从 Kate 中启动控制台命令，强大的搜索和替换插件，以及一个预览插件，可以渲染 MD、HTML 甚至 SVG 文件。支持通过交换文件在系统崩溃时恢复数据，带参数提示的自动补全，同时支持 [LSP (Language Server Protocol)](https://microsoft.github.io/language-server-protocol/) 以获得更为强大的补全。
```

File: docs/tools/editor/kate.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+![](images/kate-2-windows-install-2.png)
+
+其中形如 `kate-21.04.0-1-1272-windows-msvc2019_64-cl-sideload.appx` 格式的是供手动安装的商店版本，`kate-21.04.0-1-1272-windows-msvc2019_64-cl.exe` 格式的则是安装包，下载双击即可。
+
+## 用法与功能
+
+### 交换文件防止数据丢失
+
+类似 Vim，Kate 会将未保存的更改写入一个交换文件（一般是原文件名前面加点后面加 `.kate-swp`），如果遭遇断电或程序崩溃，您下次启动时不会丢失未保存的更改。
```
Comment (ksyx) [inline docs/tools/editor/kate.md:23]:
```suggestion
类似 Vim，Kate 会将未保存的更改写入一个交换文件（一般是原文件名前面加点后面加 `.kate-swp`），如果遭遇断电或程序崩溃等意外，下次启动时不会丢失未保存的更改。
```

File: docs/tools/editor/kate.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+### 代码高亮
+
+Kate 支持三百余种语言的语法高亮。一般来说，Kate 可以自动地选择对应的语言进行语法高亮，不过偶尔也有错误的时候，这时候可以点击最右下角的按钮，选择正确的语言。
+
+#### 自己编写语法高亮文件
+
+尽管 Kate 支持超过三百种语言的语法高亮，但是仍不免有语言未被覆盖到，此时可以自己动手编写语法高亮文件。
+Kate 自身自带的文件位于[这里](https://kate-editor.org/syntax/)，语法可参照 [Working with Syntax Highlighting](https://docs.kde.org/trunk5/en/kate/katepart/highlight.html)，编写好的文件根据[这里](https://github.com/KDE/syntax-highlighting#syntax-definition-files)放置。[这里](https://github.com/CoelacanthusHex/dotfiles/blob/80a913cc5b90d7878eb0ed77b8df2d9b97926272/kate/.local/share/katepart5/syntax/pam_env.xml)有我编写的一个可供参考。
```
Comment (ksyx) [inline docs/tools/editor/kate.md:32]: `这里` 会掉 SEO 😢 ，考虑更好的表述？

File: docs/tools/editor/kate.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+Kate 可以自动识别当前文件使用的是什么编码，如果识别错误，可以点击右下角倒数第二个按钮，选择正确的编码。
+
+同时，Kate 也可以自动识别当前文件使用的行尾符，如果识别错误，可以点击 `工具`→`行尾` / `Tool`→`End of line` 选择正确的行尾符。
+
+### 查找与替换
+
+依次单击`编辑`→`查找`（快捷键 <kbd>Ctrl</kbd>+<kbd>F</kbd>）即可打开“查找”页面。依次单击`编辑`→`替换`（快捷键 <kbd>Ctrl</kbd>+<kbd>R</kbd>）即可打开“查找与替换”页面。同时，点击左下角 `搜索与替换` 也可打开“查找与替换”页面。
+
+具体操作和其他编辑器并无太大差别，但是支持一些额外的特性，比如说：
```
Comment (ksyx) [inline docs/tools/editor/kate.md:48]:
```suggestion
具体操作和其他编辑器并无太大差别，但是支持一些额外的特性，例如：
```

File: docs/tools/editor/kate.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+            "url": "https://github.com/rust-lang/rls",
+            "highlightingModeRegex": "^Rust$"
+        }
+    }
+}
+```
+
+其中 `server` 里的每一项代表一种语言，在这个语言里，`command` 代表启动 LSP Server 所使用的命令，`command` 是一个数组，是所需要执行的命令以空格分词的结果；`url` 是 LSP 的网址；`rootIndicationFileNames` 是用于确定项目根目录的文件；`highlightingModeRegex` 则匹配某种语法高亮的名字，以确定使用哪个 LSP；如果存在 `use` 项，则代表使用 `use` 项对应的语言的配置。
+
+填写于 `设置`→`配置 Kate`→`LSP 客户端`→`用户服务器设置`。
```
Comment (ksyx) [inline docs/tools/editor/kate.md:125]:
```suggestion
该配置项位于 `设置`→`配置 Kate`→`LSP 客户端`→`用户服务器设置`。
```

**Review by ksyx — COMMENTED** [review]
thx

**Review by ksyx — APPROVED** [review]
thx

**Comment by CoelacanthusHex** [comment]
CI 怎么爆了……

**Comment by CoelacanthusHex** [comment]
呜呜呜，咱的语文太差了

**Review by ksyx — APPROVED** [review]

**Review by Konano — APPROVED** [review]

**Pull request merged**